rotkehlchen/accounting/importer/json.py

```diff
@@ -0,0 +1,96 @@
+import json
+import logging
+from pathlib import Path
+from typing import Tuple
+
+from marshmallow import EXCLUDE, ValidationError
+from pysqlcipher3 import dbapi2 as sqlcipher
+
+from rotkehlchen.api.v1.schemas import (
+    HistoryBaseEntrySchema,
+    IgnoredActionsModifySchema,
+    ModifiableSettingsSchema,
+)
+from rotkehlchen.db.dbhandler import DBHandler
+from rotkehlchen.db.history_events import DBHistoryEvents
+from rotkehlchen.errors.misc import InputError
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+class JSONImporter:
+    def __init__(self, db: DBHandler) -> None:
+        self.db = db
+
+    def import_history_debug(self, filepath: Path) -> Tuple[bool, str]:
+        """Imports the user events, settings & ignored actions identifiers for debugging."""
+        try:
+            with open(filepath) as f:
+                debug_data = json.load(f)
+        except (PermissionError, json.JSONDecodeError) as e:
+            return False, str(e)
+
+        has_required_keys = (
+            'events' in debug_data or
+            'settings' in debug_data or
+            'ignored_events_ids' in debug_data
+        )
+        if has_required_keys is False:
+            return False, 'import history data contains invalid keys'
+        has_required_types = (
+            isinstance(debug_data['events'], list) or
+            isinstance(debug_data['settings'], dict) or
+            isinstance(debug_data['ignored_events_ids'], dict)
+        )
+        if has_required_types is False:
+            return False, 'import history data contains invalid data types'
+
+        # add history events.
+        log.debug('Trying to add history events')
+        history_db = DBHistoryEvents(self.db)
+        try:
```

Comment:
The issue with that is that we'd miss some events. Rather than importing an incomplete export data, we reject the whole thing. You might a different opinion about this and I'd be open to it.

Reply:
I think it is okay to leave it as is since these are just 2 different approaches and I can't say for sure which is better. But there should be no problem with skipped events since you have a `return` statement anyway which finishes the execution of a function no matter where it is - inside or outside a loop